From 39c143bffcefc2a170c0f0421197f3475f6ab6a9 Mon Sep 17 00:00:00 2001
From: Michael Dege <michael.dege@renesas.com>
Date: Thu, 16 Jan 2025 09:15:34 +0100
Subject: [PATCH 156/157] arm64 dts renesas r8a779f0 Add Native device support

In SDK package, UIO devices were enabled by default, Native devices
(such as DMAE device) that were conflicted to UIO device.

The patch will create new DT to support enable Native devices and
disabled those UIO devices.
For convience to use:
- "r8a779f0-spider.dtb": UIO devices enable, Native devices disable.
- "r8a779f0-spider-native.dtb": UIO devices disable, Native devices enable.

Signed-off-by: Tam Nguyen <tam.nguyen.xa@renesas.com>
Signed-off-by: Michael Dege <michael.dege@renesas.com>

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index dfebff203561..1a3ebb5b2fb0 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -74,6 +74,7 @@ dtb-$(CONFIG_ARCH_R8A779A0) += r8a779a0-falcon.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk.dtb
 
 dtb-$(CONFIG_ARCH_R8A779F0) += r8a779f0-spider.dtb r8a779f0-spider-eth-2500.dtb
+dtb-$(CONFIG_ARCH_R8A779F0) += r8a779f0-spider-native.dtb
 dtb-$(CONFIG_ARCH_R8A779F0) += r8a779f0-s4sk.dtb
 dtb-$(CONFIG_ARCH_R8A779F0) += r8a779f0-vc4.dtb r8a779f0-vc4-ctrl-domain.dtb
 dtb-$(CONFIG_ARCH_R8A779F0) += r8a779f0-vc4V1.dtb r8a779f0-vc4V1-ctrl-domain.dtb
diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-spider-native.dts b/arch/arm64/boot/dts/renesas/r8a779f0-spider-native.dts
new file mode 100644
index 000000000000..25042d5f40f2
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-spider-native.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0 or MIT)
+/*
+ * Device Tree Source support Native device
+ * for the Spider CPU and Breakout boards
+ *
+ * Copyright (C) 2022 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a779f0-spider.dts"
+#include "r8a779f0-native.dtsi"
+
+/ {
+	model = "Renesas Spider Native driver based on r8a779f0";
+};
-- 
2.25.1

